src/sentry/static/sentry/app/components/avatarSettings.jsx

```diff
@@ -0,0 +1,515 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import AlertActions from '../actions/alertActions';
+import ApiMixin from '../mixins/apiMixin';
+import LoadingError from '../components/loadingError';
+import LoadingIndicator from '../components/loadingIndicator';
+import {t} from '../locale';
+
+
+const AvatarCropper = React.createClass({
+  propTypes: {
+    user: React.PropTypes.object.isRequired,
+    updateDataUrlState: React.PropTypes.func.isRequired,
+    savedDataUrl: React.PropTypes.string
+  },
+
+  mixins: [ApiMixin],
+
+  getInitialState() {
+    return {
+      error: null,
+      mousePosition: {
+        pageX: null,
+        pageY: null
+      },
+      resizeDimensions: {
+        top: 0,
+        left: 0,
+        size: 0
+      }
+    };
+  },
+
+  MIN_DIMENSION: 256,
+
+  MAX_DIMENSION: 1024,
+
+  onChange(ev) {
+    let file = ev.target.files[0];
+    if (!/^image\//.test(file.type)) {
+      this.setState({error: 'That is not a supported file type.'});
+      return;
+    }
+    this.state.objectURL && window.URL.revokeObjectURL(this.state.objectURL);
```

Comment:
Did you verify that this works with Edge? Only asking because MDN's compatibility says Edge support is unknown ("?"), and we've seen regressions in Edge features before. I *assume* it's okay, but probably worth taking a minute with BrowserStack to verify if you haven't already.

https://developer.mozilla.org/en-US/docs/Web/API/URL#Browser_compatibility